spec/models/user.rb

```diff
@@ -28,10 +29,19 @@ class User
   validates :password, confirmation: { message: 'Password confirmation must match given password' }
   validates :provider_uid, presence: true
   validates :locale, inclusion: { in: ->(_user) { %i[en ru] } }
+  if Mongoid::Compatibility::Version.mongoid3_or_older?
+    validates :provider_type, inclusion: { in: ->(user) { user.allowed_provider_types } }
+  else
+    validates :provider_type, inclusion: { in: :allowed_provider_types }
+  end
```

Comment:
Thanks @rodrigopinto for the feedback.

Situation here is little bit different then in [mass assignment matcher](https://github.com/mongoid/mongoid-rspec/blob/master/lib/matchers/allow_mass_assignment.rb#L14) as Rails won't let you get to this point. It will raise an exception already when loading `User` class, like it did in tests [1]:

```
An error occurred while loading ./spec/unit/accept_nested_attributes_spec.rb.
Failure/Error: validates :provider_type, inclusion: { in: :allowed_provider_types }
ArgumentError:
  An object with the method #include? or a proc or lambda is required, and must be supplied as the :in (or :within) option of the configuration hash
# /home/travis/.rvm/gems/ruby-2.3.1/gems/activemodel-3.2.22.5/lib/active_model/validations/inclusion.rb:12:in `check_validity!'
```

so adding check to matcher wouldn't make sense.

My changes are applicable to `mongoid > 3` so there are no new tests covering mongoid3 specific cases. This are non-mongoid3 specific cases ;-)

I have made some changes so it is clearer and we don't create "fake" case for Mongoid 3: https://github.com/mongoid/mongoid-rspec/pull/216/commits/6d409d0ab72bf2dfef8c84bc1cb3a841e24e72a9 I hope that's better.

[1] https://travis-ci.org/mongoid/mongoid-rspec/jobs/376415658